### ISSUE TITLE
merging green but old PR's without rebasing cause regressions such as #10591

### ISSUE THREAD

**Issue by timotheecour** [nim-lang/Nim#10593]
/cc @LemonBoy @Araq 
* https://github.com/nim-lang/Nim/pull/9000 was green when merged [1] despite illegal test file (line above errormsg):
```
discard """
  line: 24
  errormsg: "illformed AST: "
"""
```
causing CI build failure regression #10591 once it got merged in devel (which I'm fixing in https://github.com/nim-lang/Nim/pull/10592 ; the 3rd regression I'm fixing today...)

This was odd, so I looked at the CI build: https://travis-ci.org/nim-lang/Nim/builds/430212355?utm_source=github_status&utm_medium=notification (see [2])

turns out it's running a 6 month old version of travis.yml (with allowed failure on OSX, tests not running with `nim cpp`, etc, as you can see in the travis config: https://travis-ci.org/nim-lang/Nim/builds/430212355/config)

morality:
* don't merge old PRs before either asking author to rebase (not merge!) against devel; ideally don't merge any PR for which travis/appveyor has gotten changes in the meantime
* if author in unreachable but PR looks good, consider opening a new PR based on top of that PR, as easy as `git fetch origin pull/<id>/head && git checkout FETCH_HEAD` etc and make sure original PR author is attributed by doing a `rebase + merge` instead of `squash + merge`

## note
closing/re-opening should be of no help since it'd still use old travis

[1]
![image](https://user-images.githubusercontent.com/2194784/52387759-437c3d80-2a40-11e9-94ad-c998eb5d29d3.png)

[2]
![image](https://user-images.githubusercontent.com/2194784/52387836-90f8aa80-2a40-11e9-90f3-6b79b6143f0c.png)

**Comment by narimiran** [comment]
This is not an issue, this could/should have been said via IRC/Gitter.
